spyder/utils/programs.py

```diff
@@ -202,12 +203,62 @@ def alter_subprocess_kwargs_by_platform(**kwargs):
     return kwargs
 
 
-def run_shell_command(cmdstr, **subprocess_kwargs):
+def run_shell_command(cmdstr, asynchronous=False, **subprocess_kwargs):
     """
     Execute the given shell command.
 
     Note that *args and **kwargs will be passed to the subprocess call.
 
+    If 'shell' is given in subprocess_kwargs it must be True,
+    otherwise ProgramError will be raised.
+
+    If 'executable' is not given in subprocess_kwargs, it will
+    be set to the value of the SHELL environment variable.
+
+    Note that stdin, stdout and stderr will be set by default
+    to PIPE unless specified in subprocess_kwargs.
+
+    Parameters
+    ----------
+    cmdstr : str
+        The string run as a shell command.
+    asynchronous : bool (False)
+        Whether to return a subprocess.Popen or asyncio.subprocess.Process.
+    **subprocess_kwargs : keyword arguments
+        These will be passed to subprocess.Popen.
+    """
+    popen = subprocess.Popen
+    pipe = subprocess.PIPE
+    if asynchronous:
+        popen = asyncio.create_subprocess_shell
+        pipe = asyncio.subprocess.PIPE
+
+    if 'shell' in subprocess_kwargs and not subprocess_kwargs['shell']:
+        raise ProgramError('The "shell" kwarg may be omitted, but if '
+                           'provided it must be True.')
+    else:
+        subprocess_kwargs['shell'] = True
+
+    # Don't pass SHELL to subprocess on Windows because it makes this
+    # fumction fail in Git Bash (where SHELL is declared; other Windows
+    # shells don't set it).
+    if not os.name == 'nt':
+        if 'executable' not in subprocess_kwargs:
+            subprocess_kwargs['executable'] = os.getenv('SHELL')
+
+    for stream in ['stdin', 'stdout', 'stderr']:
+        subprocess_kwargs.setdefault(stream, pipe)
+    subprocess_kwargs = alter_subprocess_kwargs_by_platform(
+        **subprocess_kwargs)
+    return popen(cmdstr, **subprocess_kwargs)
+
+
+def async_run_shell_command(cmdstr, **subprocess_kwargs):
```

Comment:
This new function doesn't seem to be used anywhere. Why is it necessary?

Reply:
It is not necessary. It must have erroneously survived an intermediate commit in which it should have been removed. Thanks for catching it!